### DISCUSSION TITLE
Is this behaving the way it should?

### DISCUSSION THREAD

**Discussion by Perfecto-always** [tailwindlabs/tailwindcss#8273] [Help] [answered]
I am using NextJs with tailwindcss. I have a custom component `Button.js` that takes color as one of the argument so that it can have consistent styling

#### Button.js
```js 
export default function Button({ children, color }) {
  return (
    <button className={`border p-4 hover:bg-${color}-100} text-${color}-600`}>
      {children}
    </button>
  );
}
```

#### index.js

```js
import Button from "../components/Button";

export default function IndexPage() {
  return (
    <div>
      <Button color="red">Hello</Button>
    </div>
  );
}
```

The problem is the that neither the **hover** nor the **text-color** is working when using this way. 
I think what I'm doing is a bad approach and must have seen on the tailwind site not to do this, but it gives a component consistent look, until and unless there is a better approach that I don't know.
Here is the [codesandbox link](https://codesandbox.io/s/gifted-matan-81hcgc?file=/pages/index.js)

**Comment by NovusOpus** [comment, marked as answer]
Yes. There's an explanation for this in the Docs: [Dynamic class names](https://tailwindcss.com/docs/content-configuration#dynamic-class-names).

**Reply by Perfecto-always** [reply]
Thanks mate, I got what I was looking for. 🙏
